api/rosetta/data_balance_integration_test.go

```diff
@@ -177,98 +201,542 @@ func TestGetBalance(t *testing.T) {
+			wantRosettaErrorDescription: "account address is not a valid hex-encoded string",
+			wantRosettaErrorDetails:     map[string]interface{}{"address": invalidAddressHex, "chain": dps.FlowTestnet.String()},
+		},
+		{
+			name: "invalid account ID",
+			request: rosetta.BalanceRequest{
+				NetworkID:  defaultNetwork(),
+				AccountID:  identifier.Account{Address: invalidAddress},
+				BlockID:    testBlock,
+				Currencies: defaultCurrency(),
+			},
+			wantStatusCode:              http.StatusUnprocessableEntity,
+			wantRosettaError:            configuration.ErrorInvalidAccount,
+			wantRosettaErrorDescription: "account address is not valid for configured chain",
+			wantRosettaErrorDetails:     map[string]interface{}{"address": invalidAddress, "chain": dps.FlowTestnet.String()},
+		},
+		{
+			name: "unknown currency requested",
+			request: rosetta.BalanceRequest{
+				NetworkID:  defaultNetwork(),
+				AccountID:  testAccount,
+				BlockID:    testBlock,
+				Currencies: []identifier.Currency{{Symbol: invalidToken, Decimals: 8}},
+			},
+			wantStatusCode:              http.StatusUnprocessableEntity,
+			wantRosettaError:            configuration.ErrorUnknownCurrency,
+			wantRosettaErrorDescription: "currency symbol has not been configured",
+			wantRosettaErrorDetails:     map[string]interface{}{"symbol": invalidToken, "decimals": uint(8)},
+		},
+		{
+			name: "invalid currency decimal count",
+			request: rosetta.BalanceRequest{
+				NetworkID:  defaultNetwork(),
+				AccountID:  testAccount,
+				BlockID:    testBlock,
+				Currencies: []identifier.Currency{{Symbol: dps.FlowSymbol, Decimals: 7}},
+			},
+			wantStatusCode:              http.StatusUnprocessableEntity,
+			wantRosettaError:            configuration.ErrorInvalidCurrency,
+			wantRosettaErrorDescription: fmt.Sprintf("currency decimals do not match configured default (default: %d)", dps.FlowDecimals),
+			wantRosettaErrorDetails:     map[string]interface{}{"symbol": dps.FlowSymbol, "decimals": uint(7)},
+		},
+		{
+			name: "invalid currency decimal count in a list of currencies",
+			request: rosetta.BalanceRequest{
+				NetworkID: defaultNetwork(),
+				AccountID: testAccount,
+				BlockID:   testBlock,
+				Currencies: []identifier.Currency{
+					{Symbol: dps.FlowSymbol, Decimals: dps.FlowDecimals},
+					{Symbol: dps.FlowSymbol, Decimals: 7},
+					{Symbol: dps.FlowSymbol, Decimals: dps.FlowDecimals},
+				},
+			},
+			wantStatusCode:              http.StatusUnprocessableEntity,
+			wantRosettaError:            configuration.ErrorInvalidCurrency,
+			wantRosettaErrorDescription: fmt.Sprintf("currency decimals do not match configured default (default: %d)", dps.FlowDecimals),
+			wantRosettaErrorDetails:     map[string]interface{}{"symbol": dps.FlowSymbol, "decimals": uint(7)},
+		},
+		// TODO: check - the error description we return is the cadence runtime error we get;
+		// should the API relay this to the client at all?
+		{
+			name: "account does not exist yet",
```

Comment:
This is not actually a correct statement. What we know is that the account doesn't have a Flow vault. We don't know if it was created or not. Could you please create an issue for this? We can look into explicitly checking an account's existence, so that we can differentiate in the future, but it's OK for now.

Reply:
Created an issue - https://github.com/optakt/flow-dps/issues/204